cmd/tempo-vulture/main.go

```diff
@@ -45,95 +43,192 @@ func init() {
 	flag.StringVar(&prometheusPath, "prometheus-path", "/metrics", "The path to publish Prometheus metrics to.")
 	flag.StringVar(&prometheusListenAddress, "prometheus-listen-address", ":80", "The address to listen on for Prometheus scrapes.")
 
-	flag.StringVar(&lokiBaseURL, "loki-base-url", "", "The base URL (scheme://hostname) at which to find loki.")
-	flag.StringVar(&lokiQuery, "loki-query", "", "The query to use to find traceIDs in Loki.")
-	flag.StringVar(&lokiUser, "loki-user", "", "The user to use for Loki basic auth.")
-	flag.StringVar(&lokiPass, "loki-pass", "", "The password to use for Loki basic auth.")
-
-	flag.StringVar(&tempoBaseURL, "tempo-base-url", "", "The base URL (scheme://hostname) at which to find tempo.")
+	flag.StringVar(&tempoQueryURL, "tempo-query-url", "", "The URL (scheme://hostname) at which to query Tempo.")
+	flag.StringVar(&tempoPushURL, "tempo-push-url", "", "The URL (scheme://hostname) at which to push traces to Tempo.")
 	flag.StringVar(&tempoOrgID, "tempo-org-id", "", "The orgID to query in Tempo")
-	flag.DurationVar(&tempoBackoffDuration, "tempo-backoff-duration", time.Second, "The amount of time to pause between tempo calls")
+	flag.DurationVar(&tempoBackoffDuration, "tempo-backoff-duration", 30*time.Second, "The amount of time to pause between Tempo calls")
+	flag.DurationVar(&tempoRetentionDuration, "tempo-retention-duration", 336*time.Hour, "The block retention that Tempo is using")
 }
 
 func main() {
 	flag.Parse()
 
-	glog.Error("Application Starting")
+	glog.Error("Tempo Vulture Starting")
 
-	testDurations := []time.Duration{
-		24 * time.Hour,
-		12 * time.Hour,
-		6 * time.Hour,
-		3 * time.Hour,
-		time.Hour,
-		30 * time.Minute,
-	}
+	startTime := time.Now().Unix()
+	ticker := time.NewTicker(tempoBackoffDuration)
+	slot := int64(tempoBackoffDuration/time.Second) * 2
```

Comment:
i'd like to drop this slot idea and just generate the directly on the timestamps.  i think this will make things easier to replicate outside the vulture.

i also think we should split the read/write tickers.  maybe we only want to generate a trace every 30 seconds, but execute a query every 5 seconds?

Reply:
> i'd like to drop this slot idea and just generate the directly on the timestamps. i think this will make things easier to replicate outside the vulture.

I removed the slot thing but from my POV the result was not great... Maybe I misunderstood something...

But, I think your interval approach is great, so I refactored the code to replicate that behavior. Now it's much cleaner and works flawlessly. Hopefully, it makes some sense!

> i also think we should split the read/write tickers. maybe we only want to generate a trace every 30 seconds, but execute a query every 5 seconds?

Done! :)